os/arch/arm/src/amebad/amebad_spi.c

```diff
@@ -0,0 +1,1229 @@
+
+	spiinfo("mode=%d\n", mode);
+
+	/* Has the mode changed? */
+
+	if (mode != priv->mode) {
+		/* Disable SPI if it is enabled */
+
+		priv->mode = mode;
+
+	}
+}
+
+/************************************************************************************
+ * Name: amebad_spi_setbits
+ *
+ * Description:
+ *   Set the number of bits per word.
+ *
+ * Input Parameters:
+ *   dev   - Device-specific state data
+ *   nbits - The number of bits requested
+ *
+ * Returned Value:
+ *   None
+ *
+ ************************************************************************************/
+
+static void amebad_spi_setbits(FAR struct spi_dev_s *dev, int nbits)
+{
+	FAR struct amebad_spidev_s *priv = (FAR struct amebad_spidev_s *)dev;
+
+	spiinfo("nbits=%d\n", nbits);
+
+	/* Has the number of bits changed? */
+
+	if (nbits != priv->nbits) {
+
+		if (nbits < 4 || nbits > 16) {
+			return;
+		}
+
+		/* Save the selection so the subsequence re-configurations will be faster */
+
+		priv->nbits = nbits;
+
+	}
+}
+
+/****************************************************************************
+ * Name: amebad_spi_hwfeatures
+ *
+ * Description:
+ *   Set hardware-specific feature flags.
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *   features - H/W feature flags
+ *
+ * Returned Value:
+ *   Zero (OK) if the selected H/W features are enabled; A negated errno
+ *   value if any H/W feature is not supportable.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SPI_HWFEATURES
+static int amebad_spi_hwfeatures(FAR struct spi_dev_s *dev,
+				amebad_spi_hwfeatures_t features)
+{
+#ifdef CONFIG_SPI_BITORDER
+	FAR struct amebad_spidev_s *priv = (FAR struct amebad_spidev_s *)dev;
+
+	spiinfo("features=%08x\n", features);
+
+	/* Other H/W features are not supported */
+
+	return -ENOSYS;
+#else
+	return -ENOSYS;
+#endif
+}
+#endif
+
+/************************************************************************************
+ * Name: amebad_spi_send
+ *
+ * Description:
+ *   Exchange one word on SPI
+ *
+ * Input Parameters:
+ *   dev - Device-specific state data
+ *   wd  - The word to send.  the size of the data is determined by the
+ *         number of bits selected for the SPI interface.
+ *
+ * Returned Value:
+ *   response
+ *
+ ************************************************************************************/
+
+static uint16_t amebad_spi_send(FAR struct spi_dev_s *dev, uint16_t wd)
+{
+	FAR struct amebad_spidev_s *priv = (FAR struct amebad_spidev_s *)dev;
+
+	uint16_t ret;
+	DEBUGASSERT(priv);
+
+	if (priv->spi_idx == MBED_SPI1) {
+
+		ret = spi_master_write(priv->spi_object, wd);
+	} else if (priv->spi_idx == MBED_SPI0) {
+
+		spi_slave_write(priv->spi_object, wd);
+	}
+
+	/* Check and clear any error flags (Reading from the SR clears the error
+	 * flags).
+	 */
+
+	spiinfo("Sent: %04x Return: %04x\n", wd, ret);
+
+	return ret;
```

Comment:
For `else if` case, there is no assignment of `ret`. In this case, `ret` is local variable, so trash value can return.
Please initialize the `ret` or assign for `else if` case.